packages/valory/skills/proposal_collector_abci/behaviours.py

```diff
@@ -183,6 +184,9 @@ def _get_updated_proposals(self) -> Generator[None, None, str]:
                 f"Could not retrieve data from Tally API. "
                 f"Received status code {response.status_code}."
             )
+            retries = self.synchronized_data.tally_api_retries + 1
+            if retries >= MAX_RETRIES:
+                return CollectActiveTallyProposalsRound.MAX_RETRIES_PAYLOAD
```

Comment:
Tests for these changes will come in another PR (we already have a ticket for it)